tecton_client/_internal/async_tecton_client.py

```diff
@@ -0,0 +1,59 @@
+import json
+from typing import Optional, Dict, Union
+from urllib.parse import urljoin
+
+import httpx
+
+from tecton_client import TectonClient
```

Comment:
circular import? Should we get this as `from tecton_client._internal.tecton_client import TectonClient`

Reply:
Generally also it feels kinda weird for the async client to subclass the sync client right? Is that a common pattern? Although the func names are the same, technically sync/async are different interfaces.

Reply:
+1 to Brian's comment about subclassing. Could the two clients derive from a common base?

Reply:
Turned into a `utils` style instead of sub-class. Makes for more code duplication but avoids conflating the two interfaces.